dfdaemon/handler/root_handler.go

```diff
@@ -65,13 +65,17 @@ func Process(w http.ResponseWriter, r *http.Request) {
 		} else {
 			log.Warnf("registry not config but url host is %s", hostIP)
 		}
+	case global.CommandLine.TrustHosts[hostIP]:
```

Comment:
In my opinion, we should pick up the value from the map before the controlling-stream enters the `swich-case`. Use approaches as below:
```
value, ok := m[xxx]
```
But, if you can find some evidence to prove that there is no risk in here, you can tell me.

Reply:
Thanks for the comments @fengzixu 

I thinks this should be fine, if the key of this map is not exist, the default value is `""`, so the value of `global.CommandLine.TrustHosts[hostIP]` is `""` here.

But there is an extreme case which is what if hostIP is `""`?  I think this should be ok, cause if the `hostIP` is `""` so that the request could not be forwarded at all.

Reply:
1. The standard way of getting value from a map in Golang is using `value, ok := m[key]`. I think we should obey this rule. 

2. As you said,  the request wasn't be forwarded when `global.CommandLine.TrustHosts[hostIP]` is `""`. But it makes the code less readable. As a reviewer, I don't know why you assign the null values to `targetURL.Host` and `targetURL.Scheme` respectively and what happens if `targetURL.Host` and `targetURL.Scheme` are `""` . However, previous logic looks like so clear:  if we don't find a valid hostIP, we will output warning info and throw the error. Although there is perhaps no impact after you modified,  we shouldn't do anything except logging if `global.CommandLine.TrustHosts[hostIP]` is `""`.

Reply:
@fengzixu 

 Thanks,  I've move "pick up map" into "default case" , and added "return" if `r.URL.Host` is "" in 
 previous code.

Reply:
@fengzixu 

Thanks for you review, i've fixed it.

1. Added "return" and `w.WriteHeader(http.StatusForbidden)` when `r.URL.Host` is still "" in previous code which already logging the host is empty condition.
2. Move "pick up map" before `switch-case` and added `case trustHost` in `switch-case`.

Please have a look, thx.